sdl2-sys/build.rs

```diff
@@ -4,83 +4,268 @@
 extern crate pkg_config;
 #[cfg(feature = "bindgen")]
 extern crate bindgen;
+#[cfg(feature="bundled")]
+extern crate cmake;
+#[cfg(feature="bundled")]
+extern crate tar;
+#[cfg(feature="bundled")]
+extern crate flate2;
+#[cfg(feature="bundled")]
+extern crate reqwest;
 
-use std::path::PathBuf;
-use std::env;
-use std::fs;
+#[macro_use]
+extern crate cfg_if;
 
-const SDL2_BUNDLED_VERSION: &str = "2.0.6";
+use std::path::{Path, PathBuf};
+use std::{io, fs, env};
+
+// corresponds to the headers that we have in sdl2-sys/SDL2-{version}
+const SDL2_HEADERS_BUNDLED_VERSION: &str = "2.0.6";
+
+// means the lastest stable version that can be downloaded from SDL2's source
+const LASTEST_SDL2_VERSION: &str = "2.0.5";
+
+#[cfg(feature="bundled")]
+fn download_to<T: io::Write>(url: &str, mut dest: T) {
+    use io::BufRead;
+
+    let resp = reqwest::get(url).expect(&format!("Failed to GET resource: {:?}", url));
+    let size = resp.headers()
+        .get::<reqwest::header::ContentLength>()
+        .map(|ct_len| **ct_len)
+        .unwrap_or(0);
+    if !resp.status().is_success() { panic!("Download request failed with status: {:?}", resp.status()) }
+    if size == 0 { panic!("Size of content was returned was 0") }
+
+    let mut src = io::BufReader::new(resp);
+    loop {
+        let n = {
+            let mut buf = src.fill_buf().unwrap();
+            dest.write_all(&mut buf).unwrap();
+            buf.len()
+        };
+        if n == 0 { break; }
+        src.consume(n);
+    }
+}
+
+#[cfg(feature = "use-pkgconfig")]
+#[cfg(feature = "static-link")]
+fn get_pkg_config() -> pkg_config::Library {
+    pkg_config::Config::new()
+        .statik(false)
```

Comment:
This seems backwards (as does the non-static version a few lines down from here).

Reply:
Oops, good catch! I don't think it affects anything though, because it still worked for me with this on linux.

Reply:
It'll still build, but you pass `static-link` it'll build a dynamic version, and without `static-link` it'll build a static version.